services/QuillLMS/spec/models/activity_spec.rb

```diff
@@ -840,4 +840,33 @@
       it { expect(subject.count).to eq(0) }
     end
   end
+
+  describe 'translated_json' do
+    subject {activity.translated_json}
+
+    let(:activity) { create(:activity)}
+
+    context 'there are translations' do
+      it 'is data + all available languages' do
+        activity.data.merge!({'landingPageHtml' => "html"})
+        activity.create_translation_mappings
+        chinese_locale = "zh-CN"
+        english_text = activity.english_texts.first
+        chinese = create(:translated_text, english_text:, locale: chinese_locale)
+        spanish = create(:translated_text, english_text:, locale: Translatable::SPANISH_LOCALE)
+        expected_data = activity.data.merge(
+          'translations' => {
+            chinese_locale => chinese.translation,
+            Translatable::SPANISH_LOCALE => spanish.translation
+          }
+        )
```

Comment:
It feels like a lot of this could be moved into `let` and `let!` calls in the context to simplify the `it` block down to just the `expect` call.

Reply:
I tried it, but unfortunately they all sort of need to happen in order (i.e. can't make the chinese text until we've created the translation mappings) and the let syntax doesn't give you the ordering you'd want.